sqlmesh/dbt/adapter.py

```diff
@@ -295,41 +302,42 @@ def execute(
         return AdapterResponse("Success"), empty_table()
 
     def resolve_schema(self, relation: BaseRelation) -> t.Optional[str]:
-        schema = self._map_table_name(relation.database, relation.schema, relation.identifier).db
-        if not schema:
-            return None
-        return schema
+        schema = self._map_table_name(self._normalize(self._relation_to_table(relation))).db
+        return schema if schema else None
 
     def resolve_identifier(self, relation: BaseRelation) -> t.Optional[str]:
-        identifier = self._map_table_name(
-            relation.database, relation.schema, relation.identifier
-        ).name
-        if not identifier:
-            return None
-        return identifier
+        identifier = self._map_table_name(self._normalize(self._relation_to_table(relation))).name
+        return identifier if identifier else None
 
-    def _map_table_name(
-        self, database: t.Optional[str], schema: t.Optional[str], identifier: t.Optional[str]
-    ) -> exp.Table:
-        name = ".".join(p for p in (database, schema, identifier) if p is not None)
+    def _map_table_name(self, table: exp.Table) -> exp.Table:
```

Comment:
why isn't normalization happen here?

Reply:
I kept normalization and mapping separate, so it's clearer what's happening; however, we could in theory have a relation to mapping method.